main.go

```diff
@@ -170,6 +169,113 @@ func extractMethodCalls(line string) *[]string {
 //computeAST go through the AST and returns
 //a astResult struct that contains all the variable declarations
 func computeAST() *astResult {
+	jsonData, err := ioutil.ReadFile("./ast/astChallenge.json")
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	var ast Ast
+	err = json.Unmarshal([]byte(jsonData), &ast)
+	if err != nil {
+		fmt.Println("BAD")
+	}
+
+	return extractVariables(&ast.Root)
+}
+
+// Explore an AST and find variable declaration
+func extractVariables(node *JsonNode) *astResult {
+	res := astResult{}
+	if node.Type == "VariableDeclaration" {
+		nameNode := findNodeWithTypeSequence(node, []string{"VariableDeclarator", "IdentifierToken"})
```

Comment:
If you knew it were possible that for more complicated types, the branches denoting Variable declarations were formed differently, is there a more general way of searching for the appropriate nodes?